File: src/main/java/itmo/is/project/mapper/trade/TradeItemMapper.java
package itmo.is.project.mapper.trade;

import itmo.is.project.dto.trade.TradeItemDto;
import itmo.is.project.mapper.EntityMapper;
import itmo.is.project.model.trade.TradeItem;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring")
public interface TradeItemMapper extends EntityMapper<TradeItemDto, TradeItem> {

    @Override
    @Mapping(source = "trade.id", target = "tradeId")
    TradeItemDto toDto(TradeItem entity);
}

File: src/main/kotlin/itmo/isproject/mapper/trade/TradeItemMapper.kt
package itmo.isproject.mapper.trade

import itmo.isproject.dto.trade.TradeItemDto
import itmo.isproject.mapper.EntityMapper
import itmo.isproject.model.trade.TradeItem
import org.mapstruct.Mapper
import org.mapstruct.Mapping

@Mapper
interface TradeItemMapper : EntityMapper<TradeItemDto, TradeItem> {

    @Mapping(source = "trade.id", target = "tradeId")
    override fun toDto(entity: TradeItem): TradeItemDto
}
